Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env bash
+#######################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# validate-da.sh
+# Description:
+#    Validate the installation and uninstallation of the Dependency Agent.
+#    Validate the enable and disable of the Dependency Agent.
+#######################################################################
+
+set -e
+set -x
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+function check_prereqs() {
+	LogMsg "Checking for preconditions"
+
+	if [[ $EUID -ne 0 ]]; then
+	   LogErr "This script must be run as root" 
+	   exit 1
+	fi
+
+	[ -d "/opt/microsoft/dependency-agent" ] && LogErr "Directory /opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/var/opt/microsoft/dependency-agent" ] && LogErr "Directory /var/opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/etc/opt/microsoft/dependency-agent" ] && LogErr "Directory /etc/opt/microsoft/dependency-agent exists." && exit 1
+	[ -d "/lib/microsoft-dependency-agent" ] && LogErr "Directory /lib/microsoft-dependency-agent exists." && exit 1
+	[ -d "/etc/init.d/microsoft-dependency-agent" ] && LogErr "Directory /etc/init.d/microsoft-dependency-agent exists." && exit 1
+
+	LogMsg "Preconditions met"
+}
+
+function download_da_linux_installer() {
+	LogMsg "Download Dependency Agent Linux installer"
+
+	curl -L https://aka.ms/dependencyagentlinux -o InstallDependencyAgent-Linux64.bin
+
+	chmod +x InstallDependencyAgent-Linux64.bin
+}
+
+function verify_install_da() {
+	./InstallDependencyAgent-Linux64.bin -vme
+
+	ret=$?
+	if [ $ret -ne 0 ]; then
+	        LogErr "Install failed with exit code ${ret}"
+	        exit 1
+	else
+	        LogMsg "Installed with exit code 0"
+	fi
+
+	[ ! -f "/opt/microsoft/dependency-agent/uninstall" ] && LogErr "/opt/microsoft/dependency-agent/uninstall does not exist." && exit 1
+	[ ! -f "/etc/init.d/microsoft-dependency-agent" ] && LogErr "/etc/init.d/microsoft-dependency-agent does not exist." && exit 1
+	[ ! -f "/var/opt/microsoft/dependency-agent/log/install.log" ] && LogErr "/var/opt/microsoft/dependency-agent/log/install.log does not exist." && exit 1
+
+	bin_version=$(./InstallDependencyAgent-Linux64.bin --version | awk '{print $5}')
+	install_log_version=$(awk "NR==3" /var/opt/microsoft/dependency-agent/log/install.log | grep 'Dependency Agent version.revision'|cut -f2 -d ":")
+	if [ "$bin_version" -ne "$install_log_version" ]; then
+	        LogErr "Version mismatch between bin version and install log version"
```

Comment:
When it failed, what were the values?

Recommendation: To make life easier for the person having to troubleshoot a test failure, include the actual values in the log message.

Reply:
Added the values in the logs